src/OrchardCore/OrchardCore.Users.Core/Authentication/CacheTicketStore.cs

```diff
@@ -0,0 +1,155 @@
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace OrchardCore.Users.Authentication;
+
+public class CacheTicketStore : ITicketStore
+{
+    private const string _keyPrefix = "ocauth-ticket";
+    private readonly IHttpContextAccessor _accessor;
+
+    public CacheTicketStore(IHttpContextAccessor accessor)
+    {
+        _accessor = accessor;
+    }
+
+    public async Task RemoveAsync(string key)
+    {
+        var cache = _accessor.HttpContext.RequestServices.GetService<IDistributedCache>();
+        var bytes = await cache.GetAsync($"{_keyPrefix}-{key}");
+        if (bytes != null && bytes.Length > 0)
+        {
+            await cache.RemoveAsync($"{_keyPrefix}-{key}");
+        }
+    }
+
+    public async Task RenewAsync(string key, AuthenticationTicket ticket)
+    {
+        var cache = _accessor.HttpContext.RequestServices.GetService<IDistributedCache>();
+        var dataProtector = _accessor.HttpContext.RequestServices.GetService<IDataProtectionProvider>()
+                                .CreateProtector($"{nameof(CacheTicketStore)}_{IdentityConstants.ApplicationScheme}");
+        var bytes = await cache.GetAsync($"{_keyPrefix}-{key}");
+        if (bytes != null && bytes.Length > 0)
+        {
+            var ticketValue = dataProtector.Protect(SerializeTicket(ticket));
+            var expiresUtc = ticket.Properties.ExpiresUtc;
+
+            var data = Deserialize(bytes);
+            // Update Last Activity Time
+            data.LastActivityUtc = DateTime.UtcNow;
+            data.ExpiresUtc = expiresUtc;
+            data.Value = ticketValue;
+            bytes = Serialize(data);
+            await cache.SetAsync($"{_keyPrefix}-{key}", bytes, new DistributedCacheEntryOptions() { AbsoluteExpiration = expiresUtc.Value });
+        }
+    }
+    public async Task<AuthenticationTicket> RetrieveAsync(string key)
+    {
+        var cache = _accessor.HttpContext.RequestServices.GetService<IDistributedCache>();
+        var bytes = await cache.GetAsync($"{_keyPrefix}-{key}");
+
+        if (bytes == null || bytes.Length == 0)
+            return null;
+        var data = Deserialize(bytes);
+        // Update Last Activity Time
+        data.LastActivityUtc = DateTime.UtcNow;
+        bytes = Serialize(data);
+        await cache.SetAsync($"{_keyPrefix}-{key}", bytes);
+
+        var dataProtector = _accessor.HttpContext.RequestServices.GetService<IDataProtectionProvider>()
+                                .CreateProtector($"{nameof(CacheTicketStore)}_{IdentityConstants.ApplicationScheme}");
+
+        var ticket = DeserializeTicket(dataProtector.Unprotect(data.Value));
+        return ticket;
+    }
+
+    public async Task<string> StoreAsync(AuthenticationTicket ticket)
+    {
+        var key = Guid.NewGuid().ToString();
+
```

Comment:
Remove extra line

Reply:
keeping a line between statement and block is more readable